c++/src/capnp/rpc-twoparty.c++

```diff
@@ -22,12 +22,13 @@
 #include "rpc-twoparty.h"
 #include "serialize-async.h"
 #include <kj/debug.h>
+#include <kj/io.h>
 
 namespace capnp {
 
 TwoPartyVatNetwork::TwoPartyVatNetwork(kj::AsyncIoStream& stream, rpc::twoparty::Side side,
                                        ReaderOptions receiveOptions)
-    : stream(stream), side(side), peerVatId(4),
+    : stream(&stream), maxFdsPerMesage(0), side(side), peerVatId(4),
```

Comment:
It seems a little dangerous to save a pointer to something that was only passed in as a reference, no? Given the lack of guarantees around the lifetime of what was passed in.

Is this just because it'd be a lot of work to update all the callers, or is there some other reason for not making this more explicit?

Reply:
The lifetime requirements haven't changed here. We just changed from storing a reference to storing a pointer (because `kj::OneOf<>` can only store reassignable types -- something I should maybe fix).

It sounds like you might be applying a Google style rule regarding when to use references vs. pointers, but KJ style doesn't buy into the same rules.

KJ almost always prefers references over pointers, because (1) the compiler forces all references to be initialized, and (2) they are not allowed to be set to null. For nullable references, we use `kj::Maybe<T&>`. KJ code only uses raw pointers in cases where the value really must be assignable, but I feel bad about it every time.

Google's style, in contrast, says basically never to use references except for const references used as an optimization to elide a copy. The alleged advantage of Google's style is that the calling code sees an `&` marker highlighting any case where the caller needs to consider lifetime, or where the callee might modify the value. That would indeed be a huge advantage if it worked! Unfortunately, my experience is that in practice, most non-trivial values end up being pointers, which means you _don't_ have to add a `&` when passing it around. So the advantage doesn't really play out in practice.

This is all a matter of opinion, of course.

As for lifetime rules, I try to document them in comments when they diverge from some common-sense baseline rules. The baseline rules are something like:
- If a constructor takes a reference, then the reference must remain valid until the destructor. (This case applies here.)
- If a function or method that returns a promise takes a reference, the reference must remain valid until the promise resolves.
- Otherwise, reference parameters need only remain valid until the function returns, unless otherwise documented.

Reply:
Much to my own surprise, it turns out that these rules are documented:

https://github.com/capnproto/capnproto/blob/master/style-guide.md#pointers-references

https://github.com/capnproto/capnproto/blob/master/style-guide.md#ownership